tools/cmd/golden-test/main.go

```diff
@@ -55,7 +55,7 @@ type BranchResult struct {
 
 // Run runs the golden tests on the base branch and the test branch and writes the summary and
 // diff to the writer.
-func Run(writer io.Writer, baseBranch, testBranch string) error {
+func Run(writer io.Writer, baseBranch, testBranch, groupingFlag string) error {
```

Comment:
nit: it would probably be better if we actually take a bool here and convert to string within this function for better type safety :)